docs/essentials/collections/threadsafe/how toAddandRemoveItems.md

```diff
@@ -1 +1,285 @@
+        public decimal Longitude { get; set; }
+        public decimal Latitude { get; set; }
+        public int[] RecentHighTemperatures { get; set; }
+
+        public CityInfo(string name, decimal longitude, decimal latitude, int[] temps)
+        {
+            Name = name;
+            lastQueryDate = DateTime.Now;
+            Longitude = longitude;
+            Latitude = latitude;
+            RecentHighTemperatures = temps;
+        }
+
+        public CityInfo()
+        {
+        }
+
+        public CityInfo(string key)
+        {
+            Name = key;
+            // MaxValue means "not initialized"
+            Longitude = Decimal.MaxValue;
+            Latitude = Decimal.MaxValue;
+            lastQueryDate = DateTime.Now;
+            RecentHighTemperatures = new int[] { 0 };
+
+        }
+        public bool Equals(CityInfo x, CityInfo y)
+        {
+            return x.Name == y.Name && x.Longitude == y.Longitude && x.Latitude == y.Latitude;
+        }
+
+        public int GetHashCode(CityInfo obj)
+        {
+            CityInfo ci = (CityInfo)obj;
+            return ci.Name.GetHashCode();
+        }
+    }
+
+    class Program
+    {
+        // Create a new concurrent dictionary.
+        static ConcurrentDictionary<string, CityInfo> cities = new ConcurrentDictionary<string, CityInfo>();
+
+        static void Main(string[] args)
+        {
+            CityInfo[] data = 
+            {
+                new CityInfo(){ Name = "Boston", Latitude = 42.358769M, Longitude = -71.057806M, RecentHighTemperatures = new int[] {56, 51, 52, 58, 65, 56,53}},
+                new CityInfo(){ Name = "Miami", Latitude = 25.780833M, Longitude = -80.195556M, RecentHighTemperatures = new int[] {86,87,88,87,85,85,86}},
+                new CityInfo(){ Name = "Los Angeles", Latitude = 34.05M, Longitude = -118.25M, RecentHighTemperatures =   new int[] {67,68,69,73,79,78,78}},
+                new CityInfo(){ Name = "Seattle", Latitude = 47.609722M, Longitude =  -122.333056M, RecentHighTemperatures =   new int[] {49,50,53,47,52,52,51}},
+                new CityInfo(){ Name = "Toronto", Latitude = 43.716589M, Longitude = -79.340686M, RecentHighTemperatures =   new int[] {53,57, 51,52,56,55,50}},
+                new CityInfo(){ Name = "Mexico City", Latitude = 19.432736M, Longitude = -99.133253M, RecentHighTemperatures =   new int[] {72,68,73,77,76,74,73}},
+                new CityInfo(){ Name = "Rio de Janiero", Latitude = -22.908333M, Longitude = -43.196389M, RecentHighTemperatures =   new int[] {72,68,73,82,84,78,84}},
+                new CityInfo(){ Name = "Quito", Latitude = -0.25M, Longitude = -78.583333M, RecentHighTemperatures =   new int[] {71,69,70,66,65,64,61}}
+            };
+
+            // Add some key/value pairs from multiple threads.
+            Task[] tasks = new Task[2];
+
+            tasks[0] = Task.Run(() =>
+            {
+                for (int i = 0; i < 2; i++)
```

Comment:
Maybe this is something to track in another issue and fix there, but I don't like the use of a magic number here to partition the work across threads filling up the dictionary.  I'd be okay with merging this for now just for the sake of getting content in the docs, though.